### PR TITLE
turn deprecation of D1-style opovers into errors

### PR DESCRIPTION
**Pull request by ghost**
As planned for 2.094.

### CODE REVIEW

**Comment by NilsLankila** [comment]
That has an unexpected side effect on opAssign.

> src/core/thread/fiber.d(729): Error: cannot implicitly convert expression `fn` of type `void function()` to `Callable`
src/core/thread/fiber.d(736): Error: cannot implicitly convert expression `dg` of type `void delegate()` to `Callable`
src/core/thread/fiber.d(729): Error: cannot implicitly convert expression `fn` of type `void function()` to `Callable`
src/core/thread/fiber.d(736): Error: cannot implicitly convert expression `dg` of type `void delegate()` to `Callable`
src/core/thread/threadbase.d(107): Error: cannot implicitly convert expression `fn` of type `void function()` to `Callable`
src/core/thread/threadbase.d(114): Error: cannot implicitly convert expression `dg` of type `void delegate()` to `Callable`
src/core/thread/threadbase.d(107): Error: cannot implicitly convert expression `fn` of type `void function()` to `Callable`
src/core/thread/threadbase.d(114): Error: cannot implicitly convert expression `dg` of type `void delegate()` to `Callable`

File: src/dmd/opover.d [hunk 1]
```diff
@@ -678,21 +662,6 @@ Expression op_overload(Expression e, Scope* sc, TOK* pop = null)
             }
             if (!s && !s_r)
             {
-                // Try the D1-style operators, deprecated
-                if (ad1 && id)
-                {
-                    s = search_function(ad1, id);
-                    if (s && id != Id.assign)
```
Comment (NilsLankila) [inline src/dmd/opover.d:685]: here...

**Comment by Geod24** [comment]
Wait, why only 6 releases ? Shouldn't it be 10 ?

**Comment by NilsLankila** [comment]
indeed...

**Pull request closed without merging**